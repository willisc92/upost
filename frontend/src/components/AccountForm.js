import React from "react";
import { connect } from "react-redux";
import { SingleDatePicker } from "react-dates";
import Select from "react-select";
import moment from "moment";
import API from "../utils/API";

class AccountForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            calendarFocused: false,
            date: moment(),
            birthDate: moment(),
            isLoaded: false,
            interests: [],
            error: undefined,
            selectedInterests: []
        };
    }

    onDateChange = (birthDate) => {
        if (birthDate) {
            this.setState(() => ({ birthDate }));
        }
    };

    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };

    onInterestChange = (selectedOption) => {
        this.setState({ selectedInterests: selectedOption });
    };

    componentDidMount() {
        API.get("interests/").then(
            (result) => {
                const interests = result.data.map((interest) => {
                    return { value: interest.interest_tag, label: interest.interest_tag };
                });
                this.setState({
                    isLoaded: true,
                    interests: interests
                });
            },

            (error) => {
                this.setState({
                    isLoaded: true,
                    error
                });
            }
        );
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit({
            formInput: e.target,
            interests: this.state.selectedInterests.map((interest) => {
                return interest.value;
            }),
            date: this.state.date,
            birthDate: this.state.birthDate
        });
    };

    render() {
        return (
            <form onSubmit={this.onSubmit}>
                <p>Username*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.username &&
                    this.props.auth.error.username.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="username" />
                <p>150 characters of fewer. Letters, digits and @/./+/-/) only.</p>
                <p>Email Address</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.email &&
                    this.props.auth.error.email.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="email" />
                <p>First Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.first_name &&
                    this.props.auth.error.first_name.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="firstName" />
                <p>Middle Name</p>
                <input type="text" name="middleName" />
                <p>Last Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.last_name &&
                    this.props.auth.error.last_name.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="lastName" />
                <p>Birth date</p>
                <SingleDatePicker
                    date={this.state.date}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={(day) => false}
                />
                <p>Country*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.country &&
                    this.props.auth.error.country.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="country" />
                <p>State*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.state &&
                    this.props.auth.error.state.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="state" />
                <p>Steet Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.street_name &&
                    this.props.auth.error.street_name.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="streetName" />
                <p>Postal Code*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.postal_code &&
                    this.props.auth.error.postal_code.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="postalCode" />
                <p>City*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.city &&
                    this.props.auth.error.city.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="city" />
                <p>Sex</p>
                <select name="sex">
                    <option>Male</option>
                    <option>Female</option>
                </select>
                <p>Phone Number</p>
                <input type="text" name="phoneNumber" />
                <p>Interests*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.interests &&
                    this.props.auth.error.interests.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <Select
                    value={this.state.selectedInterests}
                    onChange={this.onInterestChange}
                    options={this.state.interests}
                    isMulti
                    name="interests"
                />
                <p>Password</p>
                <p>Your password contain at least {this.props.minPasswordLength} characters.</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.password &&
                    this.props.auth.error.password.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="password" />
                <p>Password Confirmation*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.password &&
                    this.props.auth.error.password.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="passwordConfirmation" />
                <br />
                <button>Sign Up</button>
            </form>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        auth: state.auth
    };
};

export default connect(mapStateToProps)(AccountForm);
