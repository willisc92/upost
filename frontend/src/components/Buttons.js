import React from "react";
import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
    whiteButton: {
        color: "white"
    },
    whiteRedButton: {
        color: "white",
        backgroundColor: theme.palette.primary.main,
        margin: 5
    }
}));

const White = (WrappedComponent) => {
    const HOC = (props) => {
        const classes = useStyles();
        return <WrappedComponent {...props} className={classes.whiteButton} />;
    };

    return HOC;
};

const WhiteRed = (WrappedComponent) => {
    const HOC = (props) => {
        const classes = useStyles();
        return <WrappedComponent {...props} className={classes.whiteRedButton} />;
    };

    return HOC;
};

export const WhiteButton = White(Button);
export const WhiteRedButton = WhiteRed(Button);
