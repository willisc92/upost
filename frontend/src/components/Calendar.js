import { Calendar, momentLocalizer } from "react-big-calendar";
import moment from "moment";
import React from "react";
const localizer = momentLocalizer(moment);

export const MyCalendar = (props) => (
    <div className="calendar_container">
        <Calendar
            localizer={localizer}
            events={props.events}
            onSelectEvent={props.onSelectEvent}
            startAccessor="start"
            endAccessor="end"
            views={["month", "day", "week"]}
        />
    </div>
);
