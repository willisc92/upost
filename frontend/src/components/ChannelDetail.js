import React from "react";
import API from "../utils/API";

class ChannelDetail extends React.Component {
    state = {
        detail: {},
        isLoaded: false,
        error: null
    };

    componentDidMount() {
        API.get("channels/").then(
            (result) => {
                this.setState({
                    isLoaded: true,
                    channels: result.data
                });
            },

            (error) => {
                this.setState({
                    isLoaded: true,
                    error
                });
            }
        );
    }

    render() {
        const { error, isLoaded, items } = this.state;
        if (error) {
            return <div>Error: {error.message}</div>;
        } else if (!isLoaded) {
            return <div>Loading...</div>;
        } else {
            return (
                <div>
                    {this.state.channels.map((channel) => (
                        <div key={channel.channel_id}>
                            <h1>{channel.channel_name}</h1>
                            <span>{channel.creation_date}</span>
                        </div>
                    ))}
                </div>
            );
        }
    }
}

export default ChannelList;
