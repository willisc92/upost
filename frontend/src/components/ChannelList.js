import React from 'react';

class ChannelList extends React.Component {
    state = {
        channels: []
    };

    async componentDidMount() {
        try {
          const res = await fetch('http://127.0.0.1:8000/api/channels/');
          const channels = await res.json();
          this.setState({
            channels
          });
        } catch (e) {
          console.log(e);
        }
    }
    
    render() {
      return (
        <div>
          {this.state.channels.map((channel) => (
            <div key={channel.channel_id}>
              <h1>{channel.channel_name}</h1>
              <span>{channel.creation_date}</span>
            </div>
          ))}
        </div>
      );
    }
}

export default ChannelList; 

    