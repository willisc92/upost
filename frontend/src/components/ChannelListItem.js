import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import {
    PolaroidHeader,
    PolaroidBody,
    PolaroidSubHeader,
    PolaroidImage,
    useStyles
} from "../components/PolaroidComponents";

export const ChannelListItem = ({ channel, selected, pathName, inHorizontalMenu }) => {
    const classes = useStyles();

    return (
        <div className={inHorizontalMenu ? `menu-item ${selected ? "active" : ""}` : ""}>
            <Link
                className="link"
                to={{
                    pathname: pathName,
                    state: { channel }
                }}
            >
                <Card className={classes.card}>
                    <CardActionArea className={classes.cardActionArea}>
                        <PolaroidImage image={channel.picture} />
                        <CardContent>
                            <PolaroidHeader header={channel.channel_name} />
                            {channel.deleted_flag && (
                                <PolaroidSubHeader
                                    color="primary"
                                    subheader={`Deleted ${moment(channel.deletion_date).format("l")}`}
                                />
                            )}
                            <PolaroidBody body={channel.channel_description} />
                            <PolaroidSubHeader subheader="Updated:" />
                            <PolaroidBody body={moment(channel.last_updated).format("MMMM Do YYYY, h:mm a")} />
                        </CardContent>
                    </CardActionArea>
                </Card>
            </Link>
        </div>
    );
};

export default ChannelListItem;

export const BrowseChannelsMenu = (list, selected, deleted_flag) =>
    list
        .filter((el) => el.deleted_flag === deleted_flag)
        .map((el) => {
            return (
                <ChannelListItem
                    channel={el}
                    key={el.channel_id}
                    selected={selected}
                    pathName={`/channel/${el.channel_id}`}
                    inHorizontalMenu
                />
            );
        });
