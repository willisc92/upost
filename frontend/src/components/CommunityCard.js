import React from "react";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import {
    PolaroidHeader,
    PolaroidBody,
    PolaroidImage,
    useStyles,
    PinnedSVG,
    UnpinnedSVG
} from "../components/PolaroidComponents";

export const CommunityCard = (props) => {
    const onClick = () => {
        props.changeIsSelected(props.community.community_name);
    };

    const classes = useStyles();

    return (
        <div className="polaroid" onClick={onClick}>
            {props.community.isSelected ? (
                <PinnedSVG className={classes.icon} color="primary" />
            ) : (
                <UnpinnedSVG className={classes.icon} color="primary" />
            )}
            <Card className={classes.card} style={{ boxShadow: "0px 0px 5px rgba(0, 0, 0, 0.5)" }}>
                <CardActionArea className={classes.cardActionArea}>
                    <PolaroidImage image={props.community.image} />
                    <CardContent>
                        <PolaroidHeader header={props.community.community_name} />
                        <PolaroidBody body={props.community.description} />
                    </CardContent>
                </CardActionArea>
            </Card>
        </div>
    );
};

export default CommunityCard;
