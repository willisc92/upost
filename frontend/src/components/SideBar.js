import React from "react";
import { slide as Menu } from "react-burger-menu";
import { Link } from "react-router-dom";
import Radium from "radium";

let RadiumLink = Radium(Link);

export class SideBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            menuOpen: false
        };
    }

    handleStateChange(state) {
        this.setState({ menuOpen: state.isOpen });
    }

    closeMenu() {
        this.setState({ menuOpen: false });
    }

    render() {
        return (
            <Menu isOpen={this.state.menuOpen} onStateChange={(state) => this.handleStateChange(state)}>
                <RadiumLink id="home" className="menu-item" to="/" onClick={() => this.closeMenu()}>
                    Home
                </RadiumLink>
                <RadiumLink id="inspire_me" className="menu-item" to="/inspire_me" onClick={() => this.closeMenu()}>
                    Inspire Me
                </RadiumLink>
                <RadiumLink id="food_mood" className="menu-item" to="/food_mood" onClick={() => this.closeMenu()}>
                    Food Mood
                </RadiumLink>
                <RadiumLink id="around_me" className="menu-item" to="/around_me" onClick={() => this.closeMenu()}>
                    Around Me
                </RadiumLink>
                <RadiumLink id="my_list" className="menu-item" to="/my_list" onClick={() => this.closeMenu()}>
                    My List
                </RadiumLink>
                <RadiumLink id="my_list" className="menu-item" to="/myDeletedContent" onClick={() => this.closeMenu()}>
                    Recycle Bin
                </RadiumLink>
            </Menu>
        );
    }
}

export default SideBar;
