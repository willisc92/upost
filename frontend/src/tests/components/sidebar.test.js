import { SideBar } from "../../components/SideBar";
import { shallow } from "enzyme";
import React from "react";

test("Should render sidebar", () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper).toMatchSnapshot();
});
