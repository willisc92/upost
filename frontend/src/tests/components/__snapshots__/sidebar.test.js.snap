// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render sidebar 1`] = `
<Menu
  bodyClassName=""
  burgerBarClassName=""
  burgerButtonClassName=""
  className=""
  crossButtonClassName=""
  crossClassName=""
  disableAutoFocus={false}
  disableCloseOnEsc={false}
  htmlClassName=""
  id=""
  itemClassName=""
  itemListClassName=""
  menuClassName=""
  morphShapeClassName=""
  noOverlay={false}
  onStateChange={[Function]}
  outerContainerId=""
  overlayClassName=""
  pageWrapId=""
  styles={Object {}}
  width={300}
>
  <a
    className="menu-item"
    href="/"
    id="home"
  >
    Home
  </a>
  <a
    className="menu-item"
    href="/inspire_me"
    id="inspire_me"
  >
    Inspire Me
  </a>
  <a
    className="menu-item"
    href="/food_mood"
    id="food_mood"
  >
    Food Mood
  </a>
  <a
    className="menu-item"
    href="/around_me"
    id="around_me"
  >
    Around Me
  </a>
  <a
    className="menu-item"
    href="/my_list"
    id="my_list"
  >
    My List
  </a>
</Menu>
`;
